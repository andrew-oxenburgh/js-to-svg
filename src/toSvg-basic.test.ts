describe('circle', () => {
   const toSvg = require('./toSvg')()

   describe('rect', () => {
      test('normal method', () => {
         let json = toSvg.rect({
            width: 200,
            height: 200,
            x: 0,
            y: 0,
            fill: 'red',
            stroke: 'none'
         })
         let actual = toSvg.stringify(json)
         expect(actual).toMatchSnapshot();
      })
      test('origin and size', () => {
         let json = toSvg.rect({
            origin: {x: 0, y: 1},
            size: {width: 3, height: 7},
            fill: 'red',
            stroke: 'none'
         })
         let actual = toSvg.stringify(json)
         expect(actual).toMatchSnapshot();
      })
   })
   describe('circle', () => {
      test('normal method', () => {
         let json = toSvg.circle({
            cx: 100,
            cy: 150,
            r: 165,
            fill: 'red',
            stroke: 'none'
         });
         let actual = toSvg.stringify(json)
         expect(actual).toMatchSnapshot();
      });
      test('accept point as center', () => {
         let json = toSvg.circle({
            center: {x: 181, y: 191},
            r: 233,
            fill: 'red',
            stroke: 'none'
         });
         let actual = toSvg.stringify(json)
         expect(actual).toMatchSnapshot();
      });
   });

   describe('doc', () => {
      test('empty doc', () => {
         let json = toSvg.createSvgObject(
            [],
            {x: 1, y: 2},
            {width: 200, height: 250});
         let actual = toSvg.stringify(json)
         expect(actual).toMatchSnapshot();
      });
   });
});

