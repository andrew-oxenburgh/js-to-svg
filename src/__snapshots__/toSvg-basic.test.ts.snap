// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`circle circle accept point as center 1`] = `"<circle fill="red" r="233" stroke="none" cx="181" cy="191"/>"`;

exports[`circle circle normal method 1`] = `"<circle fill="red" cx="100" cy="150" r="165" stroke="none"/>"`;

exports[`circle doc empty doc 1`] = `"<svg width="200" height="250" viewBox="1 2 250 200" xmlns="http://www.w3.org/2000/svg"/>"`;

exports[`circle rect normal method 1`] = `"<rect fill="red" width="200" height="200" x="0" y="0" stroke="none"/>"`;

exports[`circle rect origin and size 1`] = `"<rect fill="red" stroke="none" x="0" y="1" width="3" height="7"/>"`;
